File: algorithm/sort/quick-sort/quick-sort.js
function quicksort(q) {
  if (q.length <= 1) {
    return q;
  }

  const pivot = q[0];

  const less = new Array();
  for (let i = 1; i < q.length; i++) {
    if (q[i] < pivot) {
      less.push(q[i]);
    }
  }

  const greater = new Array();
  for (let i = 1; i < q.length; i++) {
    if (q[i] >= pivot) {
      greater.push(q[i]);
    }
  }

  const qResult = quicksort(less) + [pivot] + quicksort(greater);

  return qResult;
}

const myArray = [5, 2, 9, 3, 6, 1];
const sortedArray = quicksort(myArray);
console.log(sortedArray);

File: algorithm/sort/quick-sort/quick-sort.py
def quicksort(q):
    if len(q) <= 1:
        return q
    
    pivot = q[0]
    less = [x for x in q[1:] if x < pivot]
    greater = [x for x in q[1:] if x >= pivot]
    
    return quicksort(less) + [pivot] + quicksort(greater)

# 정렬하는 배열
myArray = [5, 2, 9, 3, 6, 1]

# 퀵 소트 함수 호출
sortedArray = quicksort(myArray)

# 정렬된 배열 출력
print(sortedArray)
